examples/shop/src/store/bucket.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { createContext, useContext } from 'react';
+import { Product } from '../types';
+
+export type AddedProduct = Product & { count: number };
+
+export type AddedProducts = Map<number, AddedProduct>;
+
+type BucketContextType = {
+    addedProducts: AddedProducts;
+    addProduct: (product: Product) => void;
+    removeProduct: (product: Product) => void;
+};
+
+const BucketContext = createContext<BucketContextType>({} as BucketContextType);
+
+type BucketProviderProps = {
+    children: React.ReactNode;
+};
+
+export function BucketProvider({ children }: BucketProviderProps) {
+    const [addedProducts, setAddedProducts] = React.useState<AddedProducts>(new Map());
+
+    function addProduct(newProduct: Product) {
+        setAddedProducts((addedProducts) => {
+            if (addedProducts.has(newProduct.id)) {
+                const addedProduct = addedProducts.get(newProduct.id) as AddedProduct;
```

Comment:
It’s cleaner to add a 'if' that the value exists. (see below)

Reply:
fixed